### PR TITLE
Translate Hooks API Reference

### PR DESCRIPTION
**Pull request by bangunbagustapa**
Translated a Hooks API Reference page. This is not a big issue but i think worth to mention. 

- I think the word **Hooks** in this context refers as a name, not a verb. So it doesn't have to be italized. But in this PR i still italicized it in order to follow the style guide and based on other contributors.
- Lastly, the introduction section on hooks has different version of translation. But it's already mentioned [here](https://github.com/reactjs/id.reactjs.org/issues/2#issuecomment-546181216). So i translated it following the proposed one by @resir014 

Please kindly review my work, thanks.

### CODE REVIEW

File: content/docs/hooks-reference.md [hunk 1]
```diff
@@ -93,29 +93,29 @@ const [state, setState] = useState(() => {
...
+Perlu diperhatikan bahwa React mungkin masih perlu me-*render* lagi komponen tersebut sebelum diabaikan. Itu seharusnya tidak perlu menjadi perhatian karena React tidak perlu masuk "lebih dalam" ke dalam bagan. Jika Anda melakukan perhitungan "mahal" saat me-*render*, Anda dapat mengoptimalkannya dengan `useMemo`.
 
 ### `useEffect` {#useeffect}
 
 ```js
 useEffect(didUpdate);
 ```
 
-Accepts a function that contains imperative, possibly effectful code.
+Menerima sebuah fungsi yang berisi kode imperatif, juga mungkin kode yang efektif.
```
Comment (resir014) [inline content/docs/hooks-reference.md]:
```suggestion
Menerima sebuah fungsi yang berisi kode imperatif, juga kode yang mungkin mengandung efek samping.
```

This translation makes more sense, IMO.

File: content/docs/hooks-reference.md [hunk 2]
```diff
@@ -450,31 +451,31 @@ function FancyInput(props, ref) {
...
 ### `useLayoutEffect` {#uselayouteffect}
 
-The signature is identical to `useEffect`, but it fires synchronously after all DOM mutations. Use this to read layout from the DOM and synchronously re-render. Updates scheduled inside `useLayoutEffect` will be flushed synchronously, before the browser has a chance to paint.
+Tanda ini identik dengan `useEffect`, tetapi diaktifkan secara *synchronous* setelah semua mutasi DOM. Gunakan ini untuk membaca tata letak dari DOM dan me-*render* ulang secara *synchronous*. Pembaruan yang dijadwalkan di dalam `useLayoutEffect` akan dipenuhi secara *synchronous*, sebelum browser memiliki kesempatan untuk menggambar.
 
-Prefer the standard `useEffect` when possible to avoid blocking visual updates.
+Pilih `useEffect` standar bila memungkinkan untuk menghindari pemblokiran pembaruan visual.
 
-> Tip
+> Kiat
```
Comment (resir014) [inline content/docs/hooks-reference.md]:
```suggestion
> Tip
```

We don't need to translate "tip", I think. We'll also have to check these blockquotes sometime in the future to make these blockquote heaedings consistent.

File: content/docs/hooks-reference.md [hunk 3]
```diff
@@ -490,17 +491,17 @@ function useFriendStatus(friendID) {
 }
 ```
 
-> Tip
+> Kiat
```
Comment (resir014) [inline content/docs/hooks-reference.md]:
```suggestion
> Tip
```

Idem.

**Comment by bangunbagustapa** [comment]
Updated the translation based on suggestions @resir014

**Pull request merged**